onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1038,22 +1120,41 @@ Applies to session load, initialization, etc. Default is 0.)pbdoc")
           "Rpbdoc(Specify values of named dimensions within model inputs.)pbdoc")
       .def(
           "add_session_config_entry",
-          [](SessionOptions* options, const char* config_key, const char* config_value) -> void {
+          [](PySessionOptions* options, const char* config_key, const char* config_value) -> void {
             const Status status = AddSessionConfigEntryImpl(*options, config_key, config_value);
             if (!status.IsOK())
               throw std::runtime_error(status.ErrorMessage());
           },
           "Rpbdoc(Set a single session configuration entry as a pair of strings.)pbdoc")
       .def(
           "get_session_config_entry",
-          [](SessionOptions* options, const char* config_key) -> std::string {
+          [](PySessionOptions* options, const char* config_key) -> std::string {
             const std::string key(config_key);
             if (!HasSessionConfigEntry(*options, key))
               throw std::runtime_error("SessionOptions does not have configuration with key: " + key);
 
             return options->session_configurations.at(key);
           },
-          "Rpbdoc(Get a single session configuration value using the given configuration key.)pbdoc");
+          "Rpbdoc(Get a single session configuration value using the given configuration key.)pbdoc")
+      .def(
+          "register_custom_ops_library",
+          [&custom_op_libraries](PySessionOptions* options, const std::string& library_path)
+              -> void {
+            OrtSessionOptions s;
+
+            // Keep the CustomOpLibrary in a container so that it can be unloaded when it is safe to do so.
+            // (Transfer ownership of the shared library to the global container that will manage all referenced
+            // shared libraries as long this module of onnxruntime is in scope.)
+            custom_op_libraries.AddLibrary(onnxruntime::make_unique<CustomOpLibrary>(library_path.c_str(), s));
+
+            // reserve enough memory to hold current contents and the new incoming contents
+            options->custom_op_domains_.reserve(options->custom_op_domains_.size() + s.custom_op_domains_.size());
+            for (size_t i = 0; i < s.custom_op_domains_.size(); ++i) {
+              std::unique_ptr<OrtCustomOpDomain> ptr(s.custom_op_domains_[i]);
```

Comment:
It looks like we're taking over the ownership of the OrtCustomOpDomain objs here. How does the user who has implemented the custom ops lib know that he should not call Release on them?

Reply:
Per offline discussion, not taking ownership here - upto the shared library implementer to release all custom op domains created within it.